pkg/pillar/cmd/zedmanager/handleimagestatus.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedmanager
+
+import (
+	"github.com/lf-edge/eve/pkg/pillar/cast"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	uuid "github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func lookupImageStatusForApp(
+	ctx *zedmanagerContext, appUUID uuid.UUID,
+	imageSha string) types.ImageStatus {
+
+	imageStatusList := ctx.subImageStatus.GetAll()
+	for _, item := range imageStatusList {
+		status := cast.CastImageStatus(item)
+		if status.AppInstUUID == appUUID && status.ImageSha256 == imageSha {
+			log.Debugf("lookupImageStatusForApp: IS found. appUUID: %s, "+
+				"imageSha: %s", appUUID.String(), imageSha)
+			return status
+		}
+	}
+	log.Debugf("lookupImageStatusForApp: Image Status NOT found. appUUID: %s, "+
+		"imageSha: %s", appUUID.String(), imageSha)
+	return types.ImageStatus{}
```

Comment:
Odd and results in odd errors checks in callers which compare some uuid.
Either make the function return a pointer (and nil if not found), or make it return (struct, error).

Reply:
Hmm.. Makes sense.. Done..